apps/accounts/migrations/0055_populate_existing_permissions.py

```diff
@@ -0,0 +1,127 @@
+# Generated by Django 3.2.19 on 2023-05-31 12:38
+
+from django.apps import apps as django_apps
+from django.db import migrations
+from guardian.shortcuts import assign_perm, remove_perm
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.auth.models import Permission, Group
+from django.contrib.auth import get_user_model
+from ..permissions import manage_datacenter, manage_provider
+
+
+def _get_managed_perms(apps, db_alias):
+    Datacenter = apps.get_model("accounts", "Datacenter")
+    Hostingprovider = apps.get_model("accounts", "Hostingprovider")
+
+    dc_perm, _ = Permission.objects.using(db_alias).get_or_create(
+        codename=manage_datacenter.codename,
+        name=manage_datacenter.description,
+        content_type=ContentType.objects.get_for_model(Datacenter),
+    )
+    hp_perm, _ = Permission.objects.using(db_alias).get_or_create(
+        codename=manage_provider.codename,
+        name=manage_provider.description,
+        content_type=ContentType.objects.get_for_model(Hostingprovider),
+    )
+
+    return (hp_perm, dc_perm)
+
+
+def set_admin_global_perms(apps, schema_editor):
+    """
+    Assign global permissions (for all Hostingprovider and Datacenter objects)
+    for all members of the "admin" group.
+    """
+    db_alias = schema_editor.connection.alias
+    admin_group = Group.objects.using(db_alias).filter(name="admin").get()
+
+    for perm in _get_managed_perms(apps, db_alias):
+        assign_perm(perm, admin_group)
+
+
+def remove_admin_global_perms(apps, schema_editor):
+    """
+    Remove global permissions (for all Hostingprovider and Datacenter objects)
+    from all members of the "admin" group.
+    """
+    db_alias = schema_editor.connection.alias
+    admin_group = Group.objects.using(db_alias).filter(name="admin").get()
+
+    for perm in _get_managed_perms(apps, db_alias):
+        remove_perm(perm, admin_group)
+
+
+def _manage_permission(apps, schema_editor, manage_func):
+    """
+    Assign/Remove object-level permissions for existing
+    Hostingprovider and Datacenter objects.
+
+    Parameter `manage_func` is either: assign_perm or remove_perm from guardian.shortcuts
+    """
+    db_alias = schema_editor.connection.alias
+
+    hp_perm, dc_perm = _get_managed_perms(apps, db_alias)
+
+    # to query the objects eligible for assigning permissions we need the historical model
+    # so that we can use queries based on fields present at the moment of the migration
```

Comment:
Woud you please add a link to some explanatory information about historical models in this?

I think it's a pretty advanced topic, and future me would really appreciate knowing there was documentatino available to help understand why models don't have the properties I was expecting, or what happend to the methods I mightt otherwise expect to call.

I found this, but there maybe others of value too.

https://docs.djangoproject.com/en/4.2/topics/migrations/#historical-models

https://docs.djangoproject.com/en/4.2/topics/migrations/#historical-models